src/main/java/me/lokka30/levelledmobs/misc/Utils.java

```diff
@@ -308,4 +312,31 @@ else if (player.getWorld().getEnvironment() == World.Environment.NORMAL){
 
         return new PlayerNetherOrWorldSpawnResult(location, isNetherPortalCoord, isWorldPortalCoord);
     }
+
+    // Copied from PaperMC
+    public static long getChunkKey(LivingEntityWrapper lmEntity){
+        int _X = (int)(lmEntity.getLivingEntity().getChunk().getX()>>4), _Z = (int)(lmEntity.getLivingEntity().getChunk().getZ()>>4);
+        return (long) _X & 0xffffffffL | ((long) _Z & 0xffffffffL) << 32;
```

Comment:
what does this method do?

Reply:
convert an chunk coordinate to long
Used for store in hashmap

Reply:
Ok - we should leave a comment so others know when they are reading the code